dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -574,6 +588,33 @@ def _correct_bias_kurtosis(match_count, biased_kurtosis):
                    (biased_kurtosis + 3) - 3 * (match_count - 1))
         return kurtosis
 
+    def _estimate_mode_from_histogram(self):
+        """
+        Estimates the mode of the current data using the
+        histogram. If there are multiple modes, returns
+        K of them (where K is defined in options given, but
+        1 by default)
+
+        :return: The estimated mode of the histogram
+        :rtype: Union[float, list(float)]
+        """
+        if not self._has_histogram:
+            return np.nan
+
+        bin_counts = self._stored_histogram['histogram']['bin_counts']
+        bin_edges = self._stored_histogram['histogram']['bin_edges']
+
+        # Get the bin(s) with the highest frequency
+        highest_idxs = np.argwhere(bin_counts == bin_counts.max()).flatten()
+        if self.mode_count is not None and len(highest_idxs) > self.mode_count:
+            highest_idxs = highest_idxs[:self.mode_count]
+
+        mode = (bin_edges[highest_idxs] + bin_edges[highest_idxs + 1]) / 2
+        # return single number instead of list if only one mode
+        if len(mode) == 1:
+            mode = mode[0]
```

Comment:
hmm, problem is this changes the data type depending on result which might be confusing. maybe just leave it as list.